drivers/sensor/ina23x/ina23x.c

```diff
@@ -0,0 +1,353 @@
+
+	switch (chan) {
+	case SENSOR_CHAN_BUS_VOLTAGE:
+		ret = ina23x_reg_read(dev, INA23X_REG_BUS_VOLT, &ina23x->bus_voltage);
+		if (ret < 0) {
+			LOG_ERR("Failed to read bus voltage");
+			return ret;
+		}
+		break;
+
+	case SENSOR_CHAN_CURRENT:
+		ret = ina23x_reg_read(dev, INA23X_REG_CURRENT, &ina23x->current);
+		if (ret < 0) {
+			LOG_ERR("Failed to read current");
+			return ret;
+		}
+		break;
+
+	case SENSOR_CHAN_POWER:
+		ret = ina23x_reg_read(dev, INA23X_REG_POWER, &ina23x->power);
+		if (ret < 0) {
+			LOG_ERR("Failed to read power");
+			return ret;
+		}
+		break;
+
+	case SENSOR_CHAN_SHUNT_VOLTAGE:
+		ret = ina23x_reg_read(dev, INA23X_REG_SHUNT_VOLT,
+				      &ina23x->shunt_voltage);
+		if (ret < 0) {
+			LOG_ERR("Failed to read shunt voltage");
+			return ret;
+		}
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief sensor attribute set
+ *
+ * @return 0 for success
+ * @return -ENOTSUP for unsupported channels
+ * @return -EIO for i2c write failure
+ */
+static int ina23x_attr_set(const struct device *dev, enum sensor_channel chan,
+			   enum sensor_attribute attr,
+			   const struct sensor_value *val)
+{
+	uint16_t data = val->val1;
+
+	switch (attr) {
+	case SENSOR_ATTR_CONFIGURATION:
+		return ina23x_reg_write(dev, INA23X_REG_CONFIG, data);
+	case SENSOR_ATTR_CALIBRATION:
+		return ina23x_reg_write(dev, INA23X_REG_CALIB, data);
+	case SENSOR_ATTR_MASK:
+		return ina23x_reg_write(dev, INA23X_REG_MASK, data);
+	case SENSOR_ATTR_ALERT:
+		return ina23x_reg_write(dev, INA23X_REG_ALERT, data);
+	default:
+		LOG_ERR("INA23X attribute not supported.");
+		return -ENOTSUP;
+	}
+}
+
+/**
+ * @brief sensor attribute get
+ *
+ * @return 0 for success
+ * @return -ENOTSUP for unsupported channels
+ * @return -EIO for i2c read failure
+ */
+static int ina23x_attr_get(const struct device *dev, enum sensor_channel chan,
+			   enum sensor_attribute attr,
+			   struct sensor_value *val)
+{
+	uint16_t data;
+	int ret;
+
+	switch (attr) {
+	case SENSOR_ATTR_CONFIGURATION:
+		ret = ina23x_reg_read(dev, INA23X_REG_CONFIG, &data);
+		if (ret < 0) {
+			return ret;
+		}
+		break;
+	case SENSOR_ATTR_CALIBRATION:
+		ret = ina23x_reg_read(dev, INA23X_REG_CALIB, &data);
+		if (ret < 0) {
+			return ret;
+		}
+		break;
+	case SENSOR_ATTR_MASK:
+		ret = ina23x_reg_read(dev, INA23X_REG_MASK, &data);
+		if (ret < 0) {
+			return ret;
+		}
+		break;
+	case SENSOR_ATTR_ALERT:
+		ret = ina23x_reg_read(dev, INA23X_REG_ALERT, &data);
+		if (ret < 0) {
+			return ret;
+		}
+		break;
+	default:
+		LOG_ERR("INA23X attribute not supported.");
+		return -ENOTSUP;
+	}
+
+	val->val1 = data;
+	val->val2 = 0;
+
+	return 0;
+}
+
+/**
+ * @brief Initialize the INA23x
+ *
+ * @return 0 for success
+ * @return -EINVAL on error
+ */
+static int ina23x_init(const struct device *dev)
+{
+	const struct ina23x_config *const config = dev->config;
+	uint16_t cal;
+	int ret;
+
+	ret = ina23x_reg_write(dev, INA23X_REG_CONFIG, config->config);
```

Comment:
Init function must check for bus device readyness, same as it is done for trigger gpio.

Reply:
Done